plugins/outputs/kafka/kafka.go

```diff
@@ -377,13 +404,26 @@ func (k *Kafka) Write(metrics []telegraf.Metric) error {
 					k.Log.Error("The timestamp of the message is out of acceptable range, consider increasing broker `message.timestamp.difference.max.ms`; dropping batch")
 					return nil
 				}
-				return prodErr
+				unsentMsgs[i] = prodErr.Msg
+
+				printableMsg := getPrintableMessage(*(prodErr.Msg))
+				k.Log.Errorf("Error when sending message: %+v : message = %+v", *prodErr, printableMsg)
 			}
+
+			k.Log.Debugf("Unsent messages: %+v", unsentMsgs)
+			msgs = unsentMsgs
 		}
-		return err
+
+		k.Log.Infof("Failed to send all messages after try %d", i)
```

Comment:
Can we include the number of messages we failed to send in the batch - I think `len(unsentMsgs)` would do the trick, correct?